shaders/MembraneWaterRipple.gdshader

```diff
@@ -0,0 +1,272 @@
+            // Add secondary ripples with reduced frequency
+            if (i > SecondaryStart && i < SecondaryEnd && i % (sampleStep * 2) == 0) {
+                float secondaryDist = length(position - pathPos);
+                if (secondaryDist > 1.5 && secondaryDist < 8.0) {
+                    float secondaryPhase = secondaryDist * 2.8 - time * 3.0 + float(i);
+                    float secondaryWave = 0.15 * sin(secondaryPhase) * 
+                                          exp(-0.15 * secondaryDist) * 
+                                          exp(-timeAgo * 0.3);
+                    wave += secondaryWave;
+                }
+            }
+            
+            // Add to total with time decay
+            totalRipple += wave * exp(-timeAgo * 0.6);
+        }
+    }
+    
+    float edgeFade = 1.0;
+    if (centerDist > 6.0) {
+        edgeFade = 1.0 - (centerDist - 6.0) / 6.0;
+        edgeFade = clamp(edgeFade, 0.0, 1.0);
+    }
+    
+    // Final ripple value with speed scaling
+    return totalRipple * max(speed, 0.15) * edgeFade;
+}
+
+void vertex() {
+    // Get vertex position in model space
+    vec2 pos = VERTEX.xz;
+    
+    // Distance based parameter
+    float distFromCenter = length(pos);
+    
+    // Only apply ripple effects to vertices within a reasonable distance
+    if (distFromCenter <= 12.0) {
+        // Apply ripple simulation with movement parameters
+        float displacement = SimulateRipples(pos, MovementDirection, MovementSpeed, TimeOffset);
+        
+        // Apply height displacement
+        VERTEX.y += displacement * RippleStrength;
+        
+        // Only calculate normals for visible displacement
+        if (abs(displacement) > 0.003) {
+            float stepSize = 0.05; 
+            if (distFromCenter < 6.0) {
+                stepSize = 0.04; 
+            }
+            
+            // Simplified gradient calculation
+            float dx = (SimulateRipples(pos + vec2(stepSize, 0.0), MovementDirection, MovementSpeed, TimeOffset) - 
+                       SimulateRipples(pos - vec2(stepSize, 0.0), MovementDirection, MovementSpeed, TimeOffset)) / (2.0 * stepSize);
+            float dz = (SimulateRipples(pos + vec2(0.0, stepSize), MovementDirection, MovementSpeed, TimeOffset) - 
+                       SimulateRipples(pos - vec2(0.0, stepSize), MovementDirection, MovementSpeed, TimeOffset)) / (2.0 * stepSize);
+            
+            // Create normal from gradient
+            vec3 gradientNormal = normalize(vec3(-dx * RippleStrength, 1.0, -dz * RippleStrength));
+            
+            // Blend normal for smooth transitions
+            float blendFactor = min(1.0, abs(displacement) * 8.0);
+            NORMAL = normalize(mix(NORMAL, gradientNormal, blendFactor));
+        }
+    }
+    // For distant vertices, we don't modify VERTEX or NORMAL
+}
+
+void fragment() {
+    // Calculate lighting parameters
+    float nDotL = dot(NORMAL, normalize(vec3(0.4, 0.8, 0.3)));

```

Comment:
nDotL especially arcane to me, especially as the variable on the next line seems to have a descriptive name...

I'm not a graphics programmer so I couldn't really say anything else about the shader other than the small style things.